third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -518,14 +648,21 @@ void Pingponger::getDotPingponged() {
     // numWarps=4 doesn't need asymmetric sync, return.
     return;
   } else if (numWarps == 8) { // Pingpong between warps from the same block
-    if (gLoadOps.size() != 2 || lLoadOps.size() != 2) {
+    if (origGlobalLoadCount != 2 || origLocalLoadCount != 2) {
```

Comment:
This check is unncessary now with the check at L583?

Reply:
I'm happy to remove this check, but it leads to a change in behavior. In particular, here are two examples:

1. Real GEMM kernel: In the fp8 rowwise GEMM there is a load in the epilogue that previously would cause the pipeliner to not be executed. By removing this check it will run with the PPS (because we will now see 2 loads). I do want to enable this kernel, but we may need additional checks because see point 2.
2. Hypothetical: If you have a kernel that has 1 dot and then does an non-dot operation later that results in an additional load this could also trigger the Ping Pong Scheduler. I'm not sure if we want larger changes before enabling this.

For that reason I think it would be better to remove this check in a followup which I am planning to work on.

Reply:
I see your points. lgtm then!